### ISSUE TITLE
Performing Split on Node with Perfect Results 

### ISSUE THREAD

**Issue by sik-flow** [cerlymarco/linear-tree#14]
I have an example where it performs a split on a node a node with a loss of 0.  Take a look at the below example.  It performs a split on `node 1` (where the loss = 0).  This split does not add any value to the results and the parent node (`node 1`) already gives perfect results.  

Is this the intended behavior? Or should it not perform splits when the results are already perfect? 

<img width="332" alt="linear_tree" src="https://user-images.githubusercontent.com/22510132/142290942-c6bd0ffc-8b95-4e78-a4cf-730766c37ae7.PNG">

**Comment by cerlymarco** [comment]
Hi thanks for your feedback.

This is not the intended behavior. The splits are computed until there is utility (as you can see from the [source code](https://github.com/cerlymarco/linear-tree/blob/5d77def63f34a2ec58318e2f510622bf071ef32b/lineartree/_classes.py#L161-L164)).

I try to reproduce a similar case where I build a LinearTreeRegressor on this data:

![image](https://user-images.githubusercontent.com/36955807/142377765-69a922a4-57b7-4814-8e07-ba24c4a80ba6.png)

The fitted model results in:

![image](https://user-images.githubusercontent.com/36955807/142377850-ff4ea92a-4adb-43e9-8e21-817c1a2330fc.png)

In node1 there is no utility in splitting so it is considered as a leaf. It's evident also looking at the prediction on the training set:

![image](https://user-images.githubusercontent.com/36955807/142378057-b9812250-05b8-4cad-9b29-007d48b58a04.png)

Maybe in your case, the loss of node1 is not exactly 0 (in the plot it's rounded).

[Here](https://colab.research.google.com/drive/1V5Wz1eYTvEwzqtTRiBJqnWQc9eZPJUgM?usp=sharing) the running notebook.

If you support the project don't forget to leave a star ;-)
